CARET_trace/src/ros_trace_points.cpp

```diff
@@ -1068,5 +1116,181 @@ void ros_trace_rmw_subscription_init(
+    ipb << "," <<
+    subscription << std::endl;
+#endif
+  };
+
+  auto now = clock.now();
+
+  if (!data_container.is_assigned_rclcpp_ipb_to_subscription()){
+    data_container.assign_rclcpp_ipb_to_subscription(record);
+  }
+  check_and_run_trace_node();
+
+  data_container.store_rclcpp_ipb_to_subscription(ipb, subscription, now);
+  record(ipb, subscription, now);
+}
+
+void ros_trace_rclcpp_construct_ring_buffer(
+    const void * buffer,
+    const uint64_t capacity
+)
+{
+  static auto & context = Singleton<Context>::get_instance();
+  static auto & clock = context.get_clock();
+  static auto & data_container = context.get_data_container();
+
+  static auto record = [](
+    const void * buffer,
+    uint64_t capacity,
+    int64_t init_timestamp
+  ){
+    tracepoint(TRACEPOINT_PROVIDER, rclcpp_construct_ring_buffer,
+      buffer, capacity, init_timestamp);
+
+#ifdef DEBUG_OUTPUT
+  std::cerr << "rclcpp_construct_ring_buffer," <<
+    buffer << "," <<
+    capacity << std::endl;
+#endif
+  };
+
+  auto now = clock.now();
+
+  if (!data_container.is_assigned_rclcpp_construct_ring_buffer()) {
+    data_container.assign_rclcpp_construct_ring_buffer(record);
+  }
+
+  check_and_run_trace_node();
+
+  data_container.store_rclcpp_construct_ring_buffer(
+    buffer, capacity, now);
+  record(buffer, capacity, now);
+}
+
+void ros_trace_rclcpp_ring_buffer_enqueue(
+    const void * buffer,
+    const uint64_t index,
+    const uint64_t size,
+    const bool overwritten
+)
+{
+  static auto & context = Singleton<Context>::get_instance();
+  static auto & controller = context.get_controller();
+  static void * orig_func = dlsym(RTLD_NEXT, __func__);
+  using functionT = void (*)(const void *, const uint64_t, const uint64_t, bool);
+  if (controller.is_allowed_buffer(buffer) &&
+    context.is_recording_allowed())
+  {
+    ((functionT) orig_func)(buffer, index, size, overwritten);
+
+#ifdef DEBUG_OUTPUT
+  std::cerr << "rclcpp_ring_buffer_enqueue," <<
+    buffer << "," <<
+    index << "," <<
+    size << "," <<
+    overwritten << std::endl;
+#endif
+  }
+}
+
+void ros_trace_rclcpp_ring_buffer_dequeue(
+    const void * buffer,
+    const uint64_t index,
+    const uint64_t size
+)
+{
+  static auto & context = Singleton<Context>::get_instance();
+  static auto & controller = context.get_controller();
+  static void * orig_func = dlsym(RTLD_NEXT, __func__);
+  using functionT = void (*)(const void *, const uint64_t, const uint64_t);
+  if (controller.is_allowed_buffer(buffer) &&
+    context.is_recording_allowed())
+  {
+  ((functionT) orig_func)(buffer, index, size);
+
+#ifdef DEBUG_OUTPUT
+  std::cerr << "rclcpp_ring_buffer_dequeue," <<
+    buffer << "," <<
+    index << "," <<
+    size << "," << std::endl;
+#endif
+  }
+}
+
+void ros_trace_rclcpp_ring_buffer_clear(
+    const void * buffer
+)
+{
+  static void * orig_func = dlsym(RTLD_NEXT, __func__);
+  using functionT = void (*)(const void *);
+  ((functionT) orig_func)(buffer);
```

Comment:
Is checking by `controller.is_allowed_buffer` and `context.is_recording_allowed` not necessary for this tracepoint?

Reply:
fixed in [64a6050](https://github.com/tier4/CARET_trace/pull/150/commits/64a60504e0a91e4b28f5535f92f65f5f5422444f).